src/phpDocumentor/Descriptor/ClassDescriptor.php

```diff
@@ -40,16 +44,16 @@ class ClassDescriptor extends DescriptorAbstract implements Interfaces\ClassInte
     /** @var bool $final Whether this class is marked as final and can't be subclassed. */
     protected $final = false;
 
-    /** @var Collection $constants References to constants defined in this class. */
+    /** @var Collection<?ConstantDescriptor> $constants References to constants defined in this class. */
```

Comment:
Aj, those nullables hurt .. we should really change the assemblers to stop them from inserting nulls into these collections :disappointed:

Reply:
Well, I didn't add them at first and it didn't seem to be an issue, I only saw later that there was special cases for null that PHPStan was raising. If there is really null in there, I don't know from where. Maybe we'll discover more with higher level analysis...

Reply:
Created issue https://github.com/phpDocumentor/phpDocumentor/issues/2246 to further investigate this